src/coreclr/inc/corinfoinstructionset.h

```diff
@@ -108,7 +108,42 @@ enum CORINFO_InstructionSet
     InstructionSet_PCLMULQDQ_X64=33,
     InstructionSet_POPCNT_X64=34,
 #endif // TARGET_X86
-
+#if defined(TARGET_WASM32) || defined(TARGET_WASM64)
+    //InstructionSet_X86Base = 1,
+    //InstructionSet_SSE = 2,
+    //InstructionSet_SSE2 = 3,
+    //InstructionSet_SSE3 = 4,
+    //InstructionSet_SSSE3 = 5,
+    //InstructionSet_SSE41 = 6,
+    InstructionSet_SSE42 = 7,
```

Comment:
It does not sound right that you have to define SSE2 and AVX2 for wasm

Reply:
They were getting used for the SIMD check https://github.com/dotnet/runtimelab/blob/fdc65fccbe815a9b81ef2ddfa6ed27e0e7292210/src/coreclr/jit/compiler.h#L8038.  Wasm can support SIMD, for c++ emscripten handles things.  for LLVM I've not looked what might be necessary, but it would be good if the clrjit optimisation for SIMD that occurs before lowering, if any, was still in I suppose.  For now this method is an `assert` for Wasm and these instructions in `corinfoinstructionset.h` are removed.

Reply:
I would expect Wasm to have its own wasm-specific names for SIMD levels, and not to reuse the x86/x64 ones. ARM does not reuse the x86/x64 ones either.

Reply:
I agree. https://github.com/WebAssembly/simd isn't finalized yet, but when it is I'd expect we would look at introducing a new `System.Runtime.Intrinsics.Wasm` namespace and corresponding ISAs.

Reply:
Ok, for now I've just deleted this change.  https://github.com/dotnet/runtimelab/pull/647/files#diff-5bc0c93aea7b72e9e37d8742e7a59ffb882ff206b746494f2e1f484bc3ef47c1